.github/workflows/workflows.yml

```diff
@@ -0,0 +1,55 @@
+name: Continuous Integration / Continuous Deployment
+on:
+  push:
+    branches:
+    - module-8
+    paths-ignore:
+    - '**.md'
+  pull_request:
+    paths-ignore:
+    - '**.md'
+jobs:
+  build:
+    name: Build the image and test it...
+    runs-on: ubuntu-latest
+    steps:
+    - name: checkout stage
+      uses: actions/checkout@v3
+    - name: building the image
+      run: docker build --target test --tag test-image-v1 .
+    - name: running the test
+      run: docker run test-image-v1
+
+  dockerhub:
+    name: Build an image to docker hub...
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.ref_name == 'module-8'
+    steps:
+    - run: echo 'Publishing...'
+    - uses: actions/checkout@v2
+
+    - name: Login to docker hub...
+      run: echo ${{secrets.DOCKERHUB_TOKEN}} | docker login --username francescowang --password-stdin
+    
+    - name: Build docker image...
+      run: docker build -t francescowang/module-8 .
+    
+    - name: Push image to docker hub...
+      run: docker push francescowang/module-8
+
+  heroku:
+    name: Deploy to heroku...
+    runs-on: ubuntu-latest
+    needs: dockerhub
+    if: github.ref_name == 'module-8'
+    steps:
+    - uses: actions/checkout@v2
+    - uses: akhileshns/heroku-deploy@v3.12.12
+      with:
+        heroku_api_key: ${{secrets.HEROKU_API}}
+        heroku_app_name: 'corndel-module-8-exercise'
+        heroku_email: "francesco.wang@kpmg.co.uk"
+        usedocker: true
+    # - name: "Checking the status code of the homepage..."
+    #   run: curl -q https://corndel-module-8-exercise.herokuapp.com/
```

Comment:
I think something like `curl -o /dev/null -sf https://corndel-module-8-exercise.herokuapp.com/` should do it, but you'll need to either wait or retry in case it takes time for heroku to finish the release